internal/iam/query.go

```diff
@@ -435,6 +435,167 @@ const (
+    ),
+    user_group_roles (role_id) as (
+      select role_id
+        from group_roles
+      union
+      select role_id
+        from user_roles
+      union
+      select role_id
+        from managed_group_roles
+    ),
+    roles_with_grants (role_id, canonical_grant) as (
+        select iam_role_grant.role_id
+             ,iam_role_grant.canonical_grant
+        from iam_role_grant
+        join iam_role
+           on iam_role.public_id               = iam_role_grant.role_id
+        join iam_grant
+           on iam_grant.canonical_grant        = iam_role_grant.canonical_grant
+        where iam_role.public_id in (select role_id from user_group_roles)
+          and iam_grant.resource               = any(@resources)
+    ),
+    global_roles as (
+        select iam_role_global.public_id             as role_id
+              ,iam_role_global.scope_id              as role_scope_id
+              ,'global'                              as role_type
+              ,iam_role_global.grant_scope           as grant_scope
+              ,iam_role_global.grant_this_role_scope as grant_this_role_scope
+              ,coalesce(individual.scope_id, '')     as individual_grant_scope
+              ,roles_with_grants.canonical_grant     as canonical_grant
+        from iam_role_global
+        join roles_with_grants
+            on roles_with_grants.role_id = iam_role_global.public_id
+        join iam_scope
+            on iam_scope.public_id = iam_role_global.scope_id
+        left join iam_role_global_individual_project_grant_scope individual
+            on individual.role_id = iam_role_global.public_id
+        where (
+          iam_role_global.grant_scope = any('{ children, descendants }')
+        ) OR (
+          iam_role_global.grant_scope = 'individual' AND
+          individual.scope_id = @request_scope
+        )
+    ),
+    org_roles as (
+        select iam_role_org.public_id             as role_id
+              ,iam_role_org.scope_id              as role_scope_id
+              ,'org'                              as role_type
+              ,iam_role_org.grant_scope           as grant_scope
+              ,iam_role_org.grant_this_role_scope as grant_this_role_scope
+              ,''                                 as individual_grant_scope
+              ,roles_with_grants.canonical_grant  as canonical_grant
+        from iam_role_org
+        join roles_with_grants
+            on roles_with_grants.role_id = iam_role_org.public_id
+        left join iam_role_org_individual_grant_scope individual
+            on individual.role_id = iam_role_org.public_id
+        where (
+          iam_role_org.grant_this_role_scope = 'children'
+        ) OR (
+          iam_role_org.grant_this_role_scope = 'individual' AND
+          individual.scope_id = @request_scope
+        )
+    ),
+    project_roles as (
+        select iam_role_project.public_id        as role_id
+              ,iam_role_project.scope_id         as role_scope_id
```

Comment:
I think scope_id wiill need to be equal to `reqScope` in order to filter a resources to specific, individual projects. (e.g. filter Hosts to a project, or even -- filtering Groups down to the project level)